src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java

```diff
@@ -75,27 +75,29 @@ public void writeToStream(OutputStream outputStream) throws IOException {
         
         outputStream.write(headerString.getBytes());
         
-        if (!isExtendedDataverseMetadataMode(formatName)) {
-            outputStream.write(METADATA_START_ELEMENT.getBytes());
-
-            outputStream.flush();
-
-            if (dataset != null && formatName != null) {
-                InputStream inputStream = null;
-                try {
-                    inputStream = ExportService.getInstance().getExport(dataset, formatName);
-                } catch (ExportException ex) {
-                    inputStream = null;
-                }
-
-                if (inputStream == null) {
-                    throw new IOException("Xrecord: failed to open metadata stream.");
+        if (header.getStatus() == null) { // Deleted records should not show metadata
+            if (!isExtendedDataverseMetadataMode(formatName)) {
+                outputStream.write(METADATA_START_ELEMENT.getBytes());
+  
+                outputStream.flush();
+  
+                if (dataset != null && formatName != null) {
+                    InputStream inputStream = null;
+                    try {
+                        inputStream = ExportService.getInstance().getExport(dataset, formatName);
+                    } catch (ExportException ex) {
+                        inputStream = null;
+                    }
+    
+                    if (inputStream == null) {
+                        throw new IOException("Xrecord: failed to open metadata stream.");
+                    }
+                    writeMetadataStream(inputStream, outputStream);
                 }
-                writeMetadataStream(inputStream, outputStream);
+                outputStream.write(METADATA_END_ELEMENT.getBytes());
+            } else {
+                outputStream.write(customMetadataExtensionRef(this.dataset.getGlobalId()).getBytes());
             }
-            outputStream.write(METADATA_END_ELEMENT.getBytes());
-        } else {
-            outputStream.write(customMetadataExtensionRef(this.dataset.getGlobalId()).getBytes());
         }
         outputStream.flush();
 
```

Comment:
@juancorr thanks again for the pull request.

At standup this morning I offered to take a look but I just requested a review from @landreev because this is all code he wrote. I loaded up the code in Netbeans and here's a screenshot with some arrows pointing to specific lines that I'm wondering about:

![screen shot 2018-05-18 at 12 23 02 pm](https://user-images.githubusercontent.com/21006/40246507-fcd54e5e-5a96-11e8-8bcb-2a35e5925cd3.png)

At line 73 I'm wondering if we should put the newly added `header.getStatus()` check in this part of the code since it's where we're throwing an `IOException` already. Maybe all the error checking should go here, I mean.

At line 101 I'm wondering if we should add an `else` for the new `header.getStatus()` check. What do we want to happen? Throw an `IOException`?

At line 113 I noticed that we're already checking `header.getStatus()` (the `itemHeaderToString` method gets called at line 70). It seems like there's potential to only have one check.

Oh, I didn't add an arrow for this but in line 87 I noticed that we're using a deprecated `getInstance` method. I left a comment about this at https://github.com/IQSS/dataverse/issues/4683#issuecomment-390258441 because it feels a little more on topic there but while we're in this part of the code I'm wondering if we should fix #4683. Here's the screenshot I added there:

![screen shot 2018-05-18 at 12 15 46 pm](https://user-images.githubusercontent.com/21006/40246753-e25bc2d2-5a97-11e8-8507-95ac8cdd38d6.png)

Reply:
Hi @pdurbin and @landreev, I think that I can response to your three arrows:

  - Line 73: In OAI protocol, when a record (dataset) has been retired (deaccessioned) should be shown as deleted without metadata but with header information (you can see first record in https://edatos.consorciomadrono.es/oai?verb=ListRecords&resumptionToken=MToyMDB8MjpvcGVuYWlyZV9kYXRhfDM6fDQ6fDU6b2FpX2RhdGFjaXRl ) . An exception at this point would hide the header information.
  - Line 101: I think that an else is not needed because header information is written out in line 76.
  - Line 113: Here is written out the "status=deleted" attribute needed for the OAI protocol

Reply:
@juancorr thanks! @landreev explained some of this to me on Friday and the pull request was already merged then but it's always good to share knowledge about how all this works!